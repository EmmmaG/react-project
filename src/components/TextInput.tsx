import * as React from 'react'
import { withStyles, WithStyles } from '@material-ui/core/styles'
import Select from '@material-ui/core/Select'
import Typography from '@material-ui/core/Typography'
import TextField from '@material-ui/core/TextField'

const styles = () => ({
	inputContainer: {
		textAlign: 'left' as 'left',
		marginTop: '10px',
		marginBottom: '20px',
	},
	textField: {
		height: '35px',
		backgroundColor: 'white',
	},
	inputRoot: {
		height: '35px',
	},
	title: {
		marginBottom: '2px',
	},
})

export interface Select {
	value: string
	text: string
}

export interface Props extends WithStyles<typeof styles> {
	title: string
	onChange: (selectedValue: string) => void
	value: string
}

export class TextInput extends React.Component<Props> {
	private handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
		this.props.onChange(event.target.value)
	}

	public render() {
		const { classes, value, title } = this.props

		return (
			<div className={classes.inputContainer}>
				<Typography className={classes.title}>{title}</Typography>
				<TextField
					id={title}
					className={classes.textField}
					value={value}
					onChange={this.handleChange}
					variant='outlined'
					placeholder='eg Falcon'
					InputProps={{ classes: { root: classes.inputRoot } }}
				/>
			</div>
		)
	}
}

export default withStyles(styles)(TextInput)
