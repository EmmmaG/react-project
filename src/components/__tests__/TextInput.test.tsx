import * as React from 'react'
import { shallow } from 'enzyme'
import idObj from 'identity-obj-proxy'

import TextInput, { Props } from '../TextInput'

const props: Props = {
	title: 'search',
	onChange: jest.fn(),
	value: '',
	classes: idObj,
}

describe('TextInput', () => {
	it('renders correctly', () => {
		const wrapper = shallow(<TextInput {...props} />)
		expect(wrapper).toMatchSnapshot()
	})
})
