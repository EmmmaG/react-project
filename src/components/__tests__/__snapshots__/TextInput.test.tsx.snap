// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput renders correctly 1`] = `
<TextInput
  classes={
    Object {
      "inputContainer": "TextInput-inputContainer-1",
      "inputRoot": "TextInput-inputRoot-3",
      "textField": "TextInput-textField-2",
      "title": "TextInput-title-4",
    }
  }
  onChange={[MockFunction]}
  title="search"
  value=""
/>
`;
